adding exception for Hosted Agents (#3095)
buildkite/docs: pages/pipelines/configure/workflows/managing_priorities.md

By default, jobs are dispatched (taken from the queue and assigned to an agent) on a first-in-first-out basis. However, job priority and pipeline upload time can affect that order.

> This is not the case for Hosted Agents. Jobs are assigned and run based on the time they are triggered.

Reviewer: Add a link to the Buildkite hosted agents overview page + reword slightly.

```suggestion
This is not the case for [Buildkite hosted agents](/docs/pipelines/hosted-agents), where jobs are assigned and dispatched at the time they are executed.
```
